lib/ansible/plugins/inventory/vmware_vm_inventory.py

```diff
@@ -244,21 +289,22 @@ def check_requirements(self):
             raise AnsibleError("Missing one of the following : hostname, username, password. Please read "
                                "the documentation for more information.")
 
+        if (self.with_folders or self.search_folder) and not self.datacenter:
+            raise AnsibleError("You must set the 'datacenter' param when setting either 'search_folder' "
+                               "or 'with_folders'.")
+
     def _get_managed_objects_properties(self, vim_type, properties=None):
```

Comment:
How about using folder value as parameter in this method rather than passing to `__init__`

```python
def _get_managed_objects_properties(self, vim_type, properties=None, folder=None):
```

Reply:
I just saw your changes to this file on PR #50645. I will check in a few days how to reconcile with those changes here.